contracts/test/TestFlashBorrower.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/IERC3156FlashBorrower.sol";
+import "../interfaces/IERC3156FlashLender.sol";
+import "../ERC20/IERC20.sol";
+
+contract TestFlashBorrower is IERC3156FlashBorrower {
+    IERC20 public immutable dola;
+    IERC3156FlashLender public immutable lender;
+    uint256 public times;
+    bytes32 public constant CALLBACK_SUCCESS = keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    constructor(IERC20 _dola, IERC3156FlashLender _lender) {
+        dola = _dola;
+        lender = _lender;
+        _dola.approve(address(_lender), type(uint256).max);
+    }
+
+    function borrow(uint256 value, uint256 _times) public {
+        times = _times - 1;
+        lender.flashLoan(IERC3156FlashBorrower(this), address(dola), value, bytes("test"));
+    }
+
+    function onFlashLoan(
+        address,
+        address token,
+        uint256 amount,
+        uint256,
+        bytes calldata data
+    ) external override returns (bytes32) {
+        if (times > 0) {
+            times = times - 1;
+            lender.flashLoan(IERC3156FlashBorrower(this), token, amount, data);
+        }
```

Comment:
Whats the purpose of this segment here? Flash loaning in a flash loan?

Reply:
Yes exactly. The original code supports re-entrant loans, this was a quick way to be able to test that branch.